core/src/Revolution/Sources/modMediaSource.php

```diff
@@ -1762,40 +1761,114 @@ protected function loadFlySystem(FilesystemAdapter $adapter)
-            $allowedFileTypes = (!is_array($allowedFileTypes)) ? explode(',', $allowedFileTypes) : $allowedFileTypes;
+        // $this->xpdo->log(
+        //     modX::LOG_LEVEL_ERROR,
+        //     "\r\tmodMediaSource->checkFileType:
+        //     get class, this: " . get_class($this) . "
+        //     source id: " . $this->get('id') . "
+        //     isStaticElement? " . $this->isStaticElementFile . "
+        //     staticPathIsAbsolute? " . $this->staticPathIsAbsolute . "
+        //     ignoreMediaSource? " . $this->ignoreMediaSource . "
+        //     opt upload_files: " . $this->xpdo->getOption('upload_files') . "
+        //     opt upload_images: " . $this->xpdo->getOption('upload_images') . "
+        //     opt upload_media: " . $this->xpdo->getOption('upload_media') . "
+        //     ms opt allowedFileTypes: " . $this->getPropertyList()['allowedFileTypes'] . "
+        //     ms opt imageExtensions " . $this->getOption('imageExtensions')
+        // );
+        $allowedFileTypes = [];
+        /*
+            Static elements will first use $staticElementFileTypes,
+            then fall back to $allowedFileTypes to determine valid types.
+        */
+        if ($staticElementFileTypes = $this->xpdo->getOption('static_elements_filetypes')) {
+            $staticElementFileTypes = !is_array($staticElementFileTypes)
+                ? explode(',', $staticElementFileTypes)
+                : $staticElementFileTypes
+                ;
+        } else {
+            $staticElementFileTypes = [];
+        }
+        /*
+            $uploadFileTypes is fallback for static element files. Its list is also combined
+            with others when checking files in other contexts, such as the media/file browser.
+        */
+        if ($uploadFileTypes = $this->xpdo->getOption('upload_files')) {
+            $uploadFileTypes = !is_array($uploadFileTypes)
+                ? explode(',', $uploadFileTypes)
+                : $uploadFileTypes
+                ;
         } else {
-            $allowedFiles = $this->xpdo->getOption('upload_files')
-                ? explode(',', $this->xpdo->getOption('upload_files')) : [];
-            $allowedImages = $this->xpdo->getOption('upload_images')
-                ? explode(',', $this->xpdo->getOption('upload_images')) : [];
-            $allowedMedia = $this->xpdo->getOption('upload_media')
-                ? explode(',', $this->xpdo->getOption('upload_media')) : [];
-            $allowedFileTypes = array_unique(array_merge($allowedFiles, $allowedImages, $allowedMedia));
-            $allowedFileTypes = array_map('trim', $allowedFileTypes);
-            $this->setOption('allowedFileTypes', $allowedFileTypes);
+            $uploadFileTypes = [];
         }
 
+        $allowedStaticFileTypes = !empty($staticElementFileTypes) ? $staticElementFileTypes : $uploadFileTypes ;
+
+        if ($this->isStaticElementFile && $this->ignoreMediaSource) {
```

Comment:
A media source should have no knowledge of static elements. This is a collision of separate concerns. This needs to be handled in some other way.